aidial_rag/retrievers/colpali_retriever/colpali_model_resource.py

```
@@ -0,0 +1,131 @@
+from aidial_rag.retrievers.colpali_retriever.colpali_index_config import (
+    ColpaliIndexConfig,
+)
+from aidial_rag.retrievers.colpali_retriever.colpali_models import (
+    MODEL_NAME_TO_TYPE,
+    KnownModels,
+    load_model_and_processor,
+)
+
+DEFAULT_BATCH_SIZE = 8
+
+
+class ColpaliModelResourceConfig(BaseModel):
+    model_name: Annotated[
+        str,
+        Field(
+            default=KnownModels.COLSMOL_256M,
+            description="Model name, should be one of MODEL_NAME_TO_TYPE keys",
+        ),
+    ]
+    batch_size: Annotated[
+        int,
+        Field(
+            default=DEFAULT_BATCH_SIZE,
+            description="Batch size for processing queries and images",
+        ),
+    ]
+
+    def validate_model_name(self):
+        """Validate that model name is known"""
+        if self.model_name not in MODEL_NAME_TO_TYPE:
+            raise ValueError(
+                f"Model name '{self.model_name}' is not known. Please use one of the following: {list(MODEL_NAME_TO_TYPE.keys())}"
+            )
+
+    @model_validator(mode="after")
+    def validate_model_consistency(self):
+        """Validate that model name is known."""
+        self.validate_model_name()
+        return self
+
+
+class ColpaliModelResource:
+    """ColPali model resource, that stores model and processor"""
+
+    def __init__(
+        self,
+        model_resource_config: ColpaliModelResourceConfig | None,
+        colpali_index_config: ColpaliIndexConfig | None,
+    ):
+        self.model_resource_config: ColpaliModelResourceConfig | None = (
+            model_resource_config
+        )
+        self.model = None
+        self.device: torch.device | None = None
+        self.processor = None
+        # if both are set then we can load model
+        if (
+            colpali_index_config is not None
+            and colpali_index_config.enabled
+            and model_resource_config is not None
+        ):
```

If one of these fields is not set, we will get a created instance of a ColpaliModelResource, which is not functional.

And we have to call `_check_model_processor_device_is_set` every time when we need to use it, because it can be in 2 states: valid one and an invalid one.

The code will be easier to read and use if the ColpaliModelResourceConfig could have only valid state. I.e. if ColpaliModelResourceConfig is not set, we can just not create an instance of ColpaliModelResourceConfig at all (this can be done in app.py). And if the ColpaliModelResourceConfig, we should assume that it is in a valid state to use if needed (otherwise it should throw an exception from the constructor if something goes wrong).

---

removed invalid state and simplified logic, now it ColpaliModelResource itself can be either none or valid depending on ColpaliModelResourceConfig which is set using app config on start.